Base/Python/slicer/util.py

```diff
@@ -1372,10 +1372,12 @@ def reloadScriptedModule(moduleName):
     for item in items:
         parent.layout().removeItem(item)
 
-    # create new widget inside existing parent
-    widget = eval("reloaded_module.%s(parent)" % widgetName)
+    # Creates new widget at slicer.modules.{widgetName}.
+    # Also ensures that qSlicerScriptedLoadableModuleWidget has references to updated enter/exit/setup methods.
+    parent.setPythonSource(filePath)
```

Comment:
I realize this method name predates this PR, but I find it really confusing that what looks like a property setter has such profound side effects.  Also it's oddly named since it takes a filePath but the method name kind of implies it would take the literal source code.  Since before it was kind of an internal class that was only available in C++ I think we could expose a new name like `setSourceFilePathAndInstantiate` and deprecate the old method name.

Reply:
I agree that `setPythonSource` would mean the input is python code rather than a filepath to python code. An updated name would be appreciated from a developer perspective 🙏🏻